File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int maxdi = INT_MIN;
        int area = 0;

        for(int i = 0; i<dimensions.size(); i++){
            vector<int> d = dimensions[i];
            int di = d[0] * d[0] + d[1] * d[1];
            if(maxdi < di || (maxdi == di && area < d[0] * d[1])){
                maxdi = di;
                area = d[0] * d[1];
            }
        }

        return area;
    }
};

File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.ts
function areaOfMaxDiagonal(dimensions: number[][]): number {
    let maxdi = Number.MIN_SAFE_INTEGER;
    let area = 0;
    dimensions.map((d,i) => {
        let di = d[0] * d[0] + d[1] * d[1];
        if(maxdi < di || (maxdi == di && area < d[0] * d[1])){
            maxdi = di;
            area = d[0] * d[1];
        }
    })

    return area;
};